src/action/UsageInterval.php

```diff
@@ -51,7 +51,8 @@ public static function wholeMonth(DateTimeImmutable $time): self
     public static function withinMonth(
         DateTimeImmutable $month,
         DateTimeImmutable $start,
-        ?DateTimeImmutable $end
+        ?DateTimeImmutable $end,
+        float $fractionOfMonth
```

Comment:
The constructor took a month, start and end dates, and calculated the fraction of month. Now I need to calculate this fraction myself before calling this class? Looks like an overconstrained constructor.  I didn't get this one. Could you explain the idea?